pdp/contract/whitelist.go

```diff
@@ -0,0 +1,54 @@
+package contract
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/filecoin-project/curio/build"
+)
+
+// AllowedRecordKeepers returns the whitelist of allowed recordkeeper addresses
+// for the public PDP service based on the build type
+func AllowedRecordKeepers() []common.Address {
+	switch build.BuildType {
+	case build.BuildCalibnet:
+		return []common.Address{
+			common.HexToAddress("0xf49ba5eaCdFD5EE3744efEdf413791935FE4D4c5"),
+		}
+	case build.BuildMainnet:
+		// Add mainnet allowed contracts here when ready
+		return []common.Address{}
+	default:
+		// For other networks (debug, 2k, etc), allow any recordkeeper
+		return nil
+	}
+}
+
+// IsPublicService checks if a service label indicates a public service
+func IsPublicService(serviceLabel string) bool {
+	return serviceLabel == "public"
+}
+
+// IsRecordKeeperAllowed checks if a recordkeeper address is in the whitelist
+// Returns true if the address is allowed, or if there's no whitelist for the network
+func IsRecordKeeperAllowed(recordKeeper common.Address) bool {
+	allowedList := AllowedRecordKeepers()
+	
+	// If allowedList is nil, it means any recordkeeper is allowed
+	if allowedList == nil {
+		return true
+	}
+	
+	// If allowedList is empty (but not nil), no recordkeepers are allowed
+	if len(allowedList) == 0 {
+		return false
+	}
+	
+	// Check if the recordkeeper is in the whitelist
+	for _, allowed := range allowedList {
+		if recordKeeper == allowed {
```

Comment:
Do you know if `common.Address` is going to be happy with case-insensitive forms of the address? Should we just lower-case all addresses to avoid any issues with case sensitivity and not demand the user provide the correct validation form?

Reply:
Good question.  After looking into it the answer is that case insensitivity is handled correctly.

Tracing through common.Address everything relies in a straightforward way on goland stdlib hex encode decode.  Stdlib reverse lookup from ascii code to hex value is here: https://cs.opensource.google/go/go/+/refs/tags/go1.24.4:src/encoding/hex/hex.go;l=25 and you can see capital letters are treated correctly.